app/javascript/components/workflows/professor/exams/admin.tsx

```diff
@@ -344,6 +344,15 @@ const ShowVersion: React.FC<{
             variant="info"
             to={`/exams/${examId}/versions/${version.id}/edit`}
             className="mr-2"
+            afterNavigate={() => {
+              if (version.anyFinalized) {
```

Comment:
From your own comment in the PR: exam versions should not be deletable or structurally editable if any students have *started*, not if any students have *finished*.

The *finished* requirement should be used to decide if the Grade button is visible.